.github/workflows/test-and-push.yml

```diff
@@ -0,0 +1,49 @@
+name: Test ddocs and push to builds server
+
+on: [push, pull_request]
+
+env:
+  BUILDS_COUCH_URL: ${{ secrets.BUILDS_URL }}
+
+jobs:
+
+  test:
+    name: Test
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Use Node.js 16.x
+        uses: actions/setup-node@v1
+        with:
+         node-version: 16.x
+
+      - name: Run CouchDb in Docker
+        run: |
+          docker run -d -p 5984:5984 --name couch couchdb:2.3.1
+          until nc -z localhost 5984; do sleep 1; done
+
+      - run: npm ci
+      - name: Test
+        run: TEST_URL=http://localhost:5984 TEST_DB=builds-test npm run test
+
+
+  push:
+    needs: test
+    name: Push to builds server
+    runs-on: ubuntu-18.04
+    if: ${{ github.event_name != 'pull_request' && github.ref == 'refs/heads/master' }}
```

Comment:
`github.event_name != 'pull_request'` can safely be omitted. According to the docs, during a `pull_request` event, the `github.ref` value should not be defined and the environment variable `GITHUB_REF` will always be `refs/pull/:prNumber/merge`. We could get the base branch with `github.head_ref` or the target branch with `github.base_ref` in case we needed it  
- https://docs.github.com/en/actions/learn-github-actions/contexts#github-context
- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

Reply:
You're right! Removed the event_name check.